### PR TITLE
Implemented #713: new option: useSnapshotsRepository adding the snapshot repo to the search list

### PR DESCRIPTION
**Pull request by ajarmoniuk**
Implementing the feature request in #713. Please note: the results from consulting the snapshots repository will be appended to the regular result list instead of replacing them.

### CODE REVIEW

File: src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java [hunk 1]
```diff
@@ -335,16 +336,14 @@ private static RuleSet getRulesViaWagon( String rulesUri, Log logger, String ser
             {
                 logger.debug( "Rule set loaded" );
 
-                if ( wagon != null )
```
Comment (ajarmoniuk) [inline src/main/java/org/codehaus/mojo/versions/api/DefaultVersionsHelper.java:338]: this is always true

**Comment by ajarmoniuk** [comment]
@slachiewicz @slawekjaranowski please review

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java [hunk 2]
```diff
@@ -121,6 +121,13 @@
     @Parameter( defaultValue = "${project.remoteArtifactRepositories}", readonly = true )
     protected List<ArtifactRepository> remoteArtifactRepositories;
 
+    /**
+     * @since 1.0-alpha-3
+     */
+    @Parameter( defaultValue = "${project.distributionManagementArtifactRepository}", readonly = true )
+    protected ArtifactRepository snapshotsRepository;
```
[the remark this answers is not in the document]
Comment (ajarmoniuk) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:128]: Yes, but this is actually the case we're after here: we want to search in deployment artifact repository. :) 

Or is there maybe another way to get a hold of the snapshot repository?

[the remark this answers is not in the document]
Comment (ajarmoniuk) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:128]: Aaah, if it's not intended to be searched then let's simply close this PR. But, on the other hand, I see that there is a method in ArtifactMetadataSource to look for an artifact in deployment repos ;)

[the remark this answers is not in the document]
Comment (slachiewicz) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:128]: Project can have multiple repositories and some of them can have flag snapshot enabled.

[the remark this answers is not in the document]
Comment (ajarmoniuk) [reply src/main/java/org/codehaus/mojo/versions/AbstractVersionsUpdaterMojo.java:128]: Then I suggest to close this PR without merging.

**Pull request closed without merging**